homeassistant/components/neato/vacuum.py

```diff
@@ -99,30 +87,19 @@ async def async_setup_entry(hass, entry, async_add_entities):
     _LOGGER.debug("Adding vacuums %s", dev)
     async_add_entities(dev, True)
 
-    def neato_custom_cleaning_service(call):
-        """Zone cleaning service that allows user to change options."""
-        for robot in service_to_entities(call):
-            if call.service == SERVICE_NEATO_CUSTOM_CLEANING:
-                mode = call.data.get(ATTR_MODE)
-                navigation = call.data.get(ATTR_NAVIGATION)
-                category = call.data.get(ATTR_CATEGORY)
-                zone = call.data.get(ATTR_ZONE)
-                try:
-                    robot.neato_custom_cleaning(mode, navigation, category, zone)
-                except NeatoRobotException as ex:
-                    _LOGGER.error("Neato vacuum connection error: %s", ex)
-
-    def service_to_entities(call):
-        """Return the known devices that a service call mentions."""
-        entity_ids = extract_entity_ids(hass, call)
-        entities = [entity for entity in dev if entity.entity_id in entity_ids]
-        return entities
-
-    hass.services.async_register(
-        NEATO_DOMAIN,
-        SERVICE_NEATO_CUSTOM_CLEANING,
-        neato_custom_cleaning_service,
-        schema=SERVICE_NEATO_CUSTOM_CLEANING_SCHEMA,
+    platform = entity_platform.current_platform.get()
+    assert platform is not None
+
+    platform.async_register_entity_service(
+        "custom_cleaning",
+        {
+            vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
```

Comment:
Why do we add the entity_id key instead of relying on the helper for that?

Reply:
Probably because we didn't know that a helper is there. Could you please give an example of how the helper can be used?

Reply:
The helper will create an enrity service schema for us when we pass a dict as schema like we do here.

https://github.com/home-assistant/core/blob/78a69ef2849baf6e0be4e34fdb9beb9bcdba98e8/homeassistant/helpers/config_validation.py#L895

https://developers.home-assistant.io/docs/dev_101_services#entity-services

We can remove this line I think.